Gallery scraping issues

After completing forum scraping, gallery scraping almost immediately fails. Oddly, when I moved the contents of the `target` folder out and retried, I got a different stack trace when scraping galleries.

```
[i] Scraping galleries...
[•] Getting albums for preset 30503267 [(1/1)]
[ERROR] Error getting album 236003: You do not have access to this gallery.
[•] Getting images for album 236004 (2/4) [(1/1)]
Inserted 236004 into table 'gallery_albums'
Inserted 5 rows into table 'gallery_images'
G:\Enjin Website Download\EnjinScraper\src\util\database.ts:4
import { TableSchema } from '../interfaces/tableschema';
           ^
TypeError: Cannot read properties of undefined (reading 'length')
    at G:\Enjin Website Download\EnjinScraper\src\util\database.ts:131:42
    at new Promise (<anonymous>)
    at G:\Enjin Website Download\EnjinScraper\src\util\database.ts:130:12
    at Generator.next (<anonymous>)
    at G:\Enjin Website Download\EnjinScraper\src\util\database.ts:8:71
    at new Promise (<anonymous>)
    at __awaiter (G:\Enjin Website Download\EnjinScraper\src\util\database.ts:4:12)
    at insertRows (G:\Enjin Website Download\EnjinScraper\src\util\database.ts:153:12)
    at G:\Enjin Website Download\EnjinScraper\src\scrapers\galleries.ts:87:29
    at Generator.next (<anonymous>)
```

```
[i] Scraping galleries...
[•] Getting albums for preset 30503267 [(1/1)]
[ERROR] Error getting album 236003: You do not have access to this gallery.
[•] Getting images for album 236004 (2/4) [(1/1)]
Inserted 236004 into table 'gallery_albums'
Inserted 5 rows into table 'gallery_images'
C:\Users\Bryce\AppData\Local\npm-cache\_npx\83b135d599749a5b\node_modules\enjinscraper\dist\src\util\database.js:155
            const values = new Array(rows[0].length).fill('?').join(', ');
                                             ^

TypeError: Cannot read properties of undefined (reading 'length')
    at C:\Users\Bryce\AppData\Local\npm-cache\_npx\83b135d599749a5b\node_modules\enjinscraper\dist\src\util\database.js:155:46
    at new Promise (<anonymous>)
    at C:\Users\Bryce\AppData\Local\npm-cache\_npx\83b135d599749a5b\node_modules\enjinscraper\dist\src\util\database.js:154:16
    at Generator.next (<anonymous>)
    at C:\Users\Bryce\AppData\Local\npm-cache\_npx\83b135d599749a5b\node_modules\enjinscraper\dist\src\util\database.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\Bryce\AppData\Local\npm-cache\_npx\83b135d599749a5b\node_modules\enjinscraper\dist\src\util\database.js:4:12)
    at insertRows (C:\Users\Bryce\AppData\Local\npm-cache\_npx\83b135d599749a5b\node_modules\enjinscraper\dist\src\util\database.js:153:12)
    at C:\Users\Bryce\AppData\Local\npm-cache\_npx\83b135d599749a5b\node_modules\enjinscraper\dist\src\scrapers\galleries.js:88:49
    at Generator.next (<anonymous>)

Node.js v19.9.0
```

The new PR is tested and working, thanks for the direction!

Also, this confirms your suspicions about the gallery tags.
```
[i] Scraping galleries...
[•] Getting albums for preset 30503267 [(1/1)]
[ERROR] Error getting album 236003: You do not have access to this gallery.
[•] Getting images for album 236004 (2/4) [(1/1)]
Inserted 236004 into table 'gallery_albums'
Inserted 5 rows into table 'gallery_images'
Ignoring insertion of 0 rows into table 'gallery_tags'
[•] Getting images for album 236005 (3/4) [(1/1)]
Inserted 236005 into table 'gallery_albums'
Inserted 4 rows into table 'gallery_images'
Ignoring insertion of 0 rows into table 'gallery_tags'
[ERROR] Error getting album 236006: You do not have access to this gallery.
Inserted galleries into table 'scrapers'
[+] Finished gallery scraping
```

I still can't figure out why I wouldn't have access to my own galleries as site owner, though.

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
     "name": "enjinscraper",
-    "version": "1.0.3",
+    "version": "1.0.4",
     "description": "Scrapes an Enjin site via the Enjin API",
     "repository": "https://github.com/Kas-tle/EnjinScraper.git",
     "author": "Joshua Castle <packages@kastle.dev",

--- a/src/util/database.ts
+++ b/src/util/database.ts
@@ -128,6 +128,10 @@ export async function updateRow(database: Database, table: string, whereKey: str
 
 export async function insertRows(database: Database, table: string, rows: (string | number | boolean | null)[][]): Promise<void> {
     return new Promise((resolve, reject) => {
+        if ( !rows.length ) {
+            statusMessage(MessageType.Plain, `Ignoring insertion of 0 rows into table '${table}'`);
+            resolve();
+        }
         const values = new Array(rows[0].length).fill('?').join(', ');
         const sql = `INSERT OR REPLACE INTO ${table} VALUES (${values})`;
 

